### ISSUE TITLE
2.5.1 broken on macos

### ISSUE THREAD

**Issue by vx8** [arzzen/git-quick-stats#152]
Hi, 

currently on macos 13.5.1, if I run 2.5.1 

`git-quick-stats -c`

I receive

> Git changelogs:
> 
> date: illegal option -- d
> usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
>             [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
>             [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]

**Comment by fatihusta** [comment]
This patch should fixes date problem for MacOS 

```
--- git-quick-stats	2023-10-24 13:28:00
+++ git-quick-stats.new	2023-10-24 13:34:42
@@ -9,6 +9,18 @@
 #       manually override these every time they want to change them?
 set -o nounset
 set -o errexit
+
+format_date() {
+  local date="${1}"
+  local outf="${2}"
+  local datef="${3:-"%b %d %H:%M:%S %Y %Z"}"  # Tue Oct 24 13:34:22 2023 +0300
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    local resp="$(date -d "${date}" "+${outf}")"
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    local resp="$(date -j -f "${datef}" "${date}" "+${outf}")"
+  fi
+  printf "%s" "${resp}"
+}

 # Beginning git log date. Respects all git datetime formats
 # If $_GIT_SINCE is never set, look at the repository to find the first date.
@@ -394,7 +406,7 @@
         --date=short "${_author}" "$_since" "$_until" $_log_options $_pathspec \
         | sort -u -r | head -n $_limit \
         | while read DATE; do
-              day=$(date -d "$DATE" "+%A")
+              day=$(format_date "$DATE" "%A" "%Y-%m-%d")
               echo -e "\n[$DATE - $day]"
               GIT_PAGER=cat git -c log.showSignature=false log \
                                 --use-mailmap $_merges \
```

**Comment by arzzen** [comment]
Hi @fatihusta , thank you for the fix, I added it to this pull request #153 
For me it works on linux (ubuntu 18, 22 and debian 12), @vx8 can you try it on macos, please?

**Comment by fatihusta** [comment]
@arzzen I already tried on MacOS and it worked.
